simavr/sim/sim_core.h

```diff
@@ -113,7 +113,7 @@ static inline void avr_sreg_set(avr_t * avr, uint8_t flag, uint8_t ival)
 	if (flag == S_I) {
 		if (ival) {
 			if (!avr->sreg[S_I])
-				avr->interrupt_state = -2;
+				avr->interrupt_state = -1;
```

Comment:
I don't understand this: the interrupt *needs* 2 cycles before firing, otherwise a lot of code will fail, like the bootloaders etc, so why are you changing that?

Reply:
The same reason I've proposed fixing it in https://github.com/buserror/simavr/pull/415/files and #376. It's verifiable against firmware that does rapid enable/disables which works fine on real hardware but sleeps indefinitely without this change in SimAVR. 

I've been running this fix for quite a while on my MK404 project with no other ill effects.